src/cache/cache_las.c

```diff
@@ -699,13 +770,36 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
                 continue;
             }
 
+            /*
+             * If we encounter a modify with preceding updates that have the same timestamp and
+             * transaction id, they need to be squashed into a single lookaside entry to avoid
+             * displacing each other.
+             */
+            if (upd->type == WT_UPDATE_MODIFY && upd->next != NULL &&
+              upd->start_ts == upd->next->start_ts && upd->txnid == upd->next->txnid) {
+                tmp_upd = upd;
+                WT_ERR(__las_squash_modifies(session, cursor, &upd));
+                if (tmp_upd->prepare_state == WT_PREPARE_INPROGRESS)
+                    ++prepared_insert_cnt;
```

Comment:
Why do we need `tmp_upd`, can't we do the increment of `prepared_insert_cnt` before calling `__las_squash_modifies`?

Reply:
If we fail to squash, I don't want to increment `prepared_insert_cnt`.

Reply:
Under what conditions will we fail to squash?

I think we've decided there is something to squash before we call `__las_squash_modifies`, right?

And on error we'll ignore `prepared_insert_cnt`?

Even more simply, could we yank the increment of `prepared_insert_cnt` above the clause that tests for modifies? I think it applies to all clauses after the birthmark test?

```
/* NEEDS A COMMENT */
if (upd->prepare_state == WT_PREPARE_INPROGRESS)
    ++prepared_insert_cnt;

/*  
 * If we encounter a modify with preceding updates that have the same timestamp and 
 * transaction id, they need to be squashed into a single lookaside entry to avoid
 * displacing each other.
 */  
if (upd->type == WT_UPDATE_MODIFY && upd->next != NULL &&
  upd->start_ts == upd->next->start_ts && upd->txnid == upd->next->txnid) {
    WT_ERR(__las_squash_modifies(session, cursor, &upd));
    ++insert_cnt;
    continue;
}   

cursor->set_value(cursor, upd->durable_ts, upd->prepare_state, upd->type, &las_value);

/* Using insert so we don't keep the page pinned longer than necessary. */
WT_ERR(cursor->insert(cursor));
++insert_cnt;
```

Reply:
189dfc2

My mistake. I missed that `prepared_insert_cnt` was ignored in the error case.